### PR TITLE
Turn on ParamSpec semanal by default

### PR DESCRIPTION
**Pull request by hauntsaninja**
This PR does not mean mypy supports PEP 612, but along with #10866 does do enough to
unblock https://github.com/python/mypy/pull/10862
mypy should now treat Callable[P, T] as Callable[..., T] and the
code paths with incomplete support will no longer crash (but will not do
what you'd want)

### CODE REVIEW

**Review by JukkaL — COMMENTED** [review]
This is an acceptable way to unblock the use of `ParamSpec` in typeshed, though we need real support for `ParamSpec`. 

Can you also add tests that check that `Callable[P, X]` is interpreted as `Callable[..., X]`, in a few different contexts (such as in a decorator)?

File: mypy/applytype.py [hunk 1]
```diff
@@ -19,6 +19,8 @@ def get_target_type(
     skip_unsatisfied: bool
 ) -> Optional[Type]:
     # TODO(shantanu): fix for ParamSpecDef
+    if not isinstance(tvar, TypeVarDef):
```
Comment (JelleZijlstra) [inline mypy/applytype.py:22]: Maybe check specifically for ParamSpecDef so we still hit the assertion if we unexpectedly get some other type. (Same applies to the next change.)

**Comment by hauntsaninja** [comment]
@JukkaL added a test! :-) If you have a minute, could you also take a look at #9951 ? It's been open for several months and simplifies some of the PEP 612 adjacent code

**Review by msullivan — APPROVED** [review]
I agree with Jelle's suggestion. So make that change if you agree, but otherwise good to go

**Pull request merged**